docs/src/domains/steps/_index.md

```diff
@@ -80,55 +127,129 @@ The CDN should already have been configured in advance to point to Platform.sh a
-After adding the line the file will look something like:
+<--->
+---
+title=Using the CLI
+file=none
+highlight=false
+---
 
-![Hosts File](/images/config-files/hosts-file.png "0.4")
+Run the following command:
 
-Alternatively, there's an add-on for Firefox
-that allows you to dynamically switch DNS IP addresses without modifying your `hosts` file.
+```bash
+platform domain:add -p <PROJECT_ID> <YOUR_DOMAIN>
+```
 
-* [Firefox LiveHosts add-on](https://addons.mozilla.org/en-US/firefox/addon/livehosts/)
+{{< /codetabs >}}
+
+Adding a domain to your project tells the edge layer where to route requests for your website.
+You can add multiple domains to point to your project.
+
+Adding a custom domain disables the automatically generated URLs (`main-def456-abc123.eu-2.platformsh.site`).
+When a domain is added to your project,
+the `{default}` in `routes.yaml` is replaced with `<YOUR_DOMAIN>` anywhere it appears when generating routes to respond to.
+Access the original internal domain by running `platform environment:info edge_hostname -e <BRANCH_NAME>`.
 
 {{< note >}}
+If you are planning on using subdomains across multiple projects, [the setup differs slightly](subdomains.md).
+{{< /note >}}
 
-Don't put the IP address you see here, but the one you got from the ping command.
+## Result
 
-*Also, remember to remove this entry after you have configured DNS!*
+With the assumption that all caches are empty, an incoming request for `mysite.com` results in the following:
 
-{{< /note >}}
+1. Your browser asks the DNS root servers for `mysite.com`'s DNS A record (the IP address of this host).
+   It responds with "it's an alias for `www.main-def456-abc123.eu-2.platformsh.site`" (the CNAME),
+   which itself resolves to the A record with IP address `1.2.3.4` (or whatever the actual address is).
+   By default DNS requests by browsers are recursive, so there is no performance penalty for using CNAMEs.
+2. Your browser sends a request to `1.2.3.4` for domain `mysite.com`.
+3. Your router responds with an HTTP 301 redirect to `www.mysite.com` (because that's what `routes.yaml` specified).
+4. Your browser looks up `www.mysite.com` and, as above, gets an alias for `www.main-def456-abc123.eu-2.platformsh.site`, which is IP `1.2.3.4`.
+5. Your browser sends a request to `1.2.3.4` for domain `www.mysite.com`.
+   Your router passes the request through to your application which in turn responds with whatever it's supposed to do.
 
-Sometimes it can take Let's Encrypt a couple of minutes to provision the certificate the first time.
-This is normal, and only means the first deploy after enabling a domain may take longer than usual.
-Setting the CNAME record with your DNS provider first helps to minimize that disruption.
+On subsequent requests, your browser knows to connect to `1.2.3.4` for domain `www.mysite.com` and skips the rest.
+The entire process takes only a few milliseconds.
 
-## 4. Bonus steps (Optional)
+## Bonus steps (Optional)
```

Comment:
```suggestion
## Optional: Bonus steps
```
Might help to have an intro to this section as well.

Reply:
Since this is the only bonus step, it might be even easier to keep the host file editing.